pkg/team/team_test.go

```diff
@@ -0,0 +1,219 @@
+
+	type args struct {
+		team gensql.Team
+	}
+	type want struct {
+		team gensql.TeamGetRow
+		err  error
+	}
+
+	operation := func(ctx context.Context, eventType gensql.EventType, team gensql.Team, teamClient *Client) bool {
+		switch eventType {
+		case gensql.EventTypeCreateTeam:
+			return teamClient.Create(ctx, team, logrus.NewEntry(logrus.StandardLogger()))
+		case gensql.EventTypeUpdateTeam:
+			return teamClient.Update(ctx, team, logrus.NewEntry(logrus.StandardLogger()))
+		case gensql.EventTypeDeleteTeam:
+			return teamClient.Delete(ctx, team.ID, logrus.NewEntry(logrus.StandardLogger()))
+		}
+
+		return true
+	}
+
+	teamTests := []struct {
+		name      string
+		eventType gensql.EventType
+		args      args
+		want      want
+	}{
+		{
+			name:      "Create team",
+			eventType: gensql.EventTypeCreateTeam,
+			args: args{
+				team: gensql.Team{
+					ID:    "test-team-1234",
+					Slug:  "test-team",
+					Users: []string{"user.one@nav.on", "user.two@nav.on"},
+					Owner: "dummy@nav.no",
+				},
+			},
+			want: want{
+				team: gensql.TeamGetRow{
+					ID:    "test-team-1234",
+					Slug:  "test-team",
+					Users: []string{"dummy@nav.no", "user.one@nav.on", "user.two@nav.on"},
+					Owner: "dummy@nav.no",
+				},
+				err: nil,
+			},
+		},
+		{
+			name:      "Create team slug already exists",
+			eventType: gensql.EventTypeCreateTeam,
+			args: args{
+				team: gensql.Team{
+					ID:    "already-exists-1234",
+					Slug:  "test-team",
+					Users: []string{},
+					Owner: "dummy@nav.no",
+				},
+			},
+			want: want{
+				team: gensql.TeamGetRow{
+					ID:    "test-team-1234",
+					Slug:  "test-team",
+					Users: []string{"dummy@nav.no", "user.one@nav.on", "user.two@nav.on"},
+					Owner: "dummy@nav.no",
+				},
+				err: nil,
+			},
+		},
+		{
+			name:      "Create team no users",
+			eventType: gensql.EventTypeCreateTeam,
+			args: args{
+				team: gensql.Team{
+					ID:    "other-team-1234",
+					Slug:  "other-team",
+					Users: []string{},
+					Owner: "dummy@nav.no",
+				},
+			},
+			want: want{
+				team: gensql.TeamGetRow{
+					ID:    "other-team-1234",
+					Slug:  "other-team",
+					Users: []string{"dummy@nav.no"},
+					Owner: "dummy@nav.no",
+				},
+				err: nil,
+			},
+		},
+		{
+			name:      "Update team",
+			eventType: gensql.EventTypeUpdateTeam,
+			args: args{
+				team: gensql.Team{
+					ID:    "test-team-1234",
+					Slug:  "test-team",
+					Users: []string{"new.user@nav.no"},
+				},
+			},
+			want: want{
+				team: gensql.TeamGetRow{
+					ID:    "test-team-1234",
+					Slug:  "test-team",
+					Users: []string{"dummy@nav.no", "new.user@nav.no"},
+					Owner: "dummy@nav.no",
+				},
+				err: nil,
+			},
+		},
+		{
+			name:      "Delete team",
+			eventType: gensql.EventTypeDeleteTeam,
+			args: args{
+				team: gensql.Team{
+					ID: "test-team-1234",
+				},
+			},
+			want: want{
+				team: gensql.TeamGetRow{
+					ID:    "",
+					Slug:  "",
+					Users: nil,
+					Owner: "",
+				},
```

Comment:
```suggestion
				team: gensql.TeamGetRow{},
```
Holder vel å skrive dette?